### DISCUSSION TITLE
New entity naming or will the naming convention be rolled back?

### DISCUSSION THREAD

**Discussion by JensenNick** [wlcrs/huawei_solar#571] [Setting up the integration] [answered]
Hi forum and a Merry Christmas to You all!
Thought I might ask here first if someone with a cascaded integration is experiencing the same "issue" as I. To me it looks like the naming convention for a cascaded setup has been altered. Previously both master and slaves were given a numeric value e.g., 0, 1, and/or 2 in the entity name. It looks like this is not the case with the latest update (version 1.3.1). Here the master is without the numeric value. Is this a bug or just a new naming convention? My master was previusly named #1.

Is this "issue" maybe related to the language settings described in this thread https://github.com/wlcrs/huawei_solar/issues/566 My profil language is English, but my system language is Danish. 

![2023-12-25_10-52-30](https://github.com/wlcrs/huawei_solar/assets/149193835/824f64bd-843c-4743-985d-71c3c2d840e7)
![2023-12-25_10-53-19](https://github.com/wlcrs/huawei_solar/assets/149193835/aef5c514-4cc5-4a7b-be33-dca3926063f5)
![2023-12-25_10-53-53](https://github.com/wlcrs/huawei_solar/assets/149193835/fbebb845-ddaf-413b-b2db-a825b04a24a5)
![2023-12-25_10-54-28](https://github.com/wlcrs/huawei_solar/assets/149193835/d7def8b3-bb2f-4255-b926-4190291c1837)

@wlcrs

**Comment by wlcrs** [comment, marked as answer]
I've had to remove the logic that suffixed each entry with '#1', '#2', etc when multiple inverters are present: it is incompatible with the HA translation logic and was causing crashes (cfr. #564).

I'm surprised that this "breaks" existing installations like shown in your screenshots. The ["unique id"](https://github.com/wlcrs/huawei_solar/blob/bbf441d3ec508fcad0c3434925b69c33855508b0/sensor.py#L774) which identifies each entity to HA didn't change, which I thought was sufficient to stay compatible with existing  installations 🤔. If anybody has any insights in what has gone wrong here: I'm all ears!

**Reply by JensenNick** [reply]
@wlcrs Thanks for the quick reply! Maybe I should clarify that the pictures posted above are from a dashboard and not the integration (created by the “Add to dashboard function”. Don’t know if that came across clearly enough. In… Settings / Devices & services / Huawei Solar… everything is looking fine.

**Comment by t785634** [comment]
Is there a way to manually change the language back to English, but only for the Huawei integration?

**Reply by JensenNick** [reply]
I believe it was suggested by @wlcrs to set the language to English while updating the integration and once updated change the language back to the one you prefer.

**Comment by t785634** [comment]
Thanks!

JensenNick ***@***.***> schrieb am So. 7. Jan. 2024 um 20:30:
